content/blog/2023/2023-03-08-code-format-with-ktlint.md

```diff
@@ -0,0 +1,125 @@
+  * **Automated checks**: We also have the capability to integrate Ktlint into our Continuous Integration pipeline, this means that every pull request or commit by developers can be automatically checked to determine whether it meets our coding standards helping us catch issues early in the development process.
+
+## Adding Ktlint to a Gradle Project
+  To add Ktlint to a Gradle project we use the [Ktlint Gradle Plugin](https://github.com/JLLeitschuh/ktlint-gradle). This plugin is easier to work with and also provides developers with commands to run in order to check and format code. Here are the steps we'll follow:
+
+### Adding the Ktlint Gradle Plugin
+  To add the Gradle plugin to our project, we add the dependencies in our root level *build.gradle*  file.
+  Let's take a look at how we achieve this:
+```groovy
+plugins {
+    id "org.jlleitschuh.gradle.ktlint" version "11.0.0"
+}
+```
+In the above code, we're using version *11.0.0* which we can replace with our version of choice.
+
+### Applying the Ktlint Plugin to Other Modules
+Let's also add Ktlint to our modules within the same project to ensure that the code in them is also checked and formatted. Note that this is only applicable if our project is a multi-module project. To achieve this task, we add the plugin in the `allprojects` block found in `build.gradle` file:
+
+```groovy
+allprojects {
+    apply plugin: "org.jlleitschuh.gradle.ktlint"
+}
+```
+## Analyzing and Formatting Code
+Let's look at which commands the Ktlint plugin provides:
+* `./gradlew ktlintCheck`:
+We use this command to tell Ktlint that it should go through our entire project and **check which files in our codebase violate the provided guidelines. After checking, this will result in a report indicating the file with errors that need to be corrected**, if there isn't any code violation the check passes.
+* `./gradlew ktlintFormat`:
+This command **automatically updates the code to follow the configured coding style**.
+
+## Adding Ktlint to a Maven Project
+In order to integrate Ktlint into our Maven project, we add the Maven plugin to our `pom.xml` file:
+
+```xml 
+<build>
+    <plugins>
+        <plugin>
+            <groupId>com.github.shyiko</groupId>
+            <artifactId>ktlint-maven-plugin</artifactId>
+            <version>RELEASE</version> 
+            <executions>
+                <execution>
+                    <id>format</id>
+                    <goals>
+                        <goal>format</goal>
+                    </goals>
+                </execution>
+            </executions>
+        </plugin>
+    </plugins>
+</build>
+```
+
+By default, after we add the Maven plugin to our project, it's using the official standard Kotlin style guide. If we want to modify and create our own rules to use in our project, we can create a `ktlint.yml` file in the root directory of our project and configure our rules in this file.
```

Comment:
Can we also use the `ktlint.yml` file with the Gradle plugin?

Reply:
Also, maybe add a warning not to customize the code format because it's much simpler just to use the official standard